src/engine/strat_engine/ns.rs

```diff
@@ -27,6 +27,8 @@ pub fn unshare_namespace() -> StratisResult<()> {
     if is_in_root_namespace()? {
         unshare(CloneFlags::CLONE_NEWNS)?;
     }
+    // unshare() seems to be working fine even if called on a PID 1 process, so the assertion below is
+    // important to trigger.
```

Comment:
My understanding from our meeting is that this assertion is actually triggered if in a container. Should the below assertion be removed?

Reply:
It depends on how we want to tackle this. I think the assertion is important for the non-container case. This assertion fails when the container does not share the host PID and runs as PID 1 because unsharing PID 1 does not seem to work.
It didn't want to change the existing behavior and simply warn the user. If we want to change this I would move `unshare_namespace()?;` under this condition https://github.com/stratis-storage/stratisd/pull/3186/files#diff-3604c519d3539d03673ac6805e179a57af023dbee55fee25322beb1be846798dR64 to avoid the failure entirely.

Reply:
I'm open to either. From my point of view, I'd almost rather exit with that error message that warn and then trigger an assertion immediately after. If we can cleanly exit, that would be preferable to me. Thoughts, @mulkieran?

Reply:
@jbaublitz: I would favor the clean exit, also.

Reply:
Thanks both, made the requested changes.